rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy

```diff
@@ -658,24 +659,40 @@ class ExecutionJob implements InterruptableJob {
     def ScheduledExecution fetchScheduledExecution(JobDataMap jobDataMap, JobExecutionContext context) {
         String seid = requireEntry(jobDataMap, "scheduledExecutionId", String)
         String projectName = requireEntry(jobDataMap, "project", String)
-        ScheduledExecution se = ScheduledExecution.findByUUID(seid).find()
-        if(se && se instanceof ScheduledExecution){
-            se.refreshOptions() //force fetch options and option values before return object
-        }
 
-        if (!se) {
-            context.getScheduler().deleteJob(context.jobDetail.key)
-            throw new ScheduledExecutionDeletedException("Failed to lookup scheduledException object from job data map: id: ${seid} , job will be unscheduled")
-        }
-        else if(!projectName.equals(se.project)){
-            context.getScheduler().deleteJob(context.jobDetail.key)
-            throw new RuntimeException("ScheduledExecution found but it does not match the original project name to schedule. Project name was : ${projectName} , Project name now : ${se.project}")
-        }
+        /**
+         * The `findByUUID` retry is a mitigation for the race condition between this quartz scheduler thread that gets
+         * ScheduledExecution ID from the memory store and the web server thread that puts the ID into the memory
+         * store AND creates the ScheduledExecution object in the DB.
+         * The issue occurs when the quartz thread starts processioning the associated job and attempts to fetch the
+         * newly (almost) created ScheduledExecution object from the DB BEFORE the DB transaction is committed by the web server thread.
+         */
+        try {
+            ScheduledExecution se = WaitUtils.<ScheduledExecution>waitFor(
+                    {ScheduledExecution.findByUUID(seid).find()},
+                    {it != null},
+                    Duration.ofSeconds(10),
+                    Duration.ofSeconds(1)
+            )
 
-        if (! se instanceof ScheduledExecution) {
-            throw new RuntimeException("JobDataMap contained invalid ScheduledExecution type: " + se.getClass().getName())
+            if(se instanceof ScheduledExecution){
+                se.refreshOptions() //force fetch options and option values before return object
+            } else {
+                // Legacy logic. Is this even possible?
+                throw new RuntimeException("JobDataMap contained invalid ScheduledExecution type: " + se.getClass().getName())
+            }
+
+            if(!projectName.equals(se.project)){
+                context.getScheduler().deleteJob(context.jobDetail.key)
```

Comment:
I'll add it to the description of the exception thrown on the next line...

Reply:
But does that exception still get logged somewhere? I see [you removed that in some places](https://github.com/rundeck/rundeck/pull/9408/files#r1815719553)

Reply:
Yes, the quartz itself logs it a few times. Verified.
However, IMO, in the codebase we should assume that someone will log it downstack -- controller error handler, thread pool error handler, etc...
It's simply not sustainable to log it "just in case".